soc/xtensa/intel_adsp/common/include/cavs_hda.h

```diff
@@ -0,0 +1,303 @@
+/* Copyright (c) 2022 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_CAVS_HDA_H
+#define ZEPHYR_INCLUDE_CAVS_HDA_H
+
+#include <arch/xtensa/cache.h>
+#include <kernel.h>
+#include <device.h>
+#include <cavs-shim.h>
+#include <cavs-mem.h>
+
+/**
+ * @brief HDA stream functionality for cAVS
+ *
+ * Provides low level calls to support cAVS HDA streams with
+ * minimal abstraction that allows testing the hardware
+ * and its demands separately from the intended DMA API
+ * usage. Notably all addresses, counts, and sizes
+ * are predefined here if not defined elsewhere to enable
+ * easier unit testing.
+ */
+
+#ifndef HDA_HOST_OUT_BASE
+#define HDA_HOST_OUT_BASE 0x72800
```

Comment:
This belong in devicetree.  Also the ifndef pattern here seems a little suspicious.  Why would you be worried about multiple definitions?  Sounds like maybe a header is missing an include guard?

Reply:
The intent here is yes, I'd move it to DTS when I go to implement the dma driver, its here because I needed something when testing. I can move this to my tests.

Reply:
I've removed the ifdefs, to remove the defines would mean I should remove the low level driver in the header I think, which I found incredibly useful.